yoti-sdk-impl/src/test/java/com/yoti/api/client/spi/remote/SimpleProfileTest.java

```diff
@@ -1,211 +1,132 @@
-        Profile profile = new SimpleProfile(list);
-
-        profile.is(null, false);
-    }
-
-    @Test(expected = IllegalArgumentException.class)
-    public void getAttribute_shouldThrowExceptionForNullAttributeName() {
-        List<Attribute> list = Collections.emptyList();
-        Profile profile = new SimpleProfile(list);
-
-        profile.getAttribute(null);
+        assertEquals(STRING_VALUE, result.getValue());
     }
 
     @Test(expected = IllegalArgumentException.class)
     public void getAttributeTyped_shouldThrowExceptionForNullAttributeName() {
-        List<Attribute> list = Collections.emptyList();
-        Profile profile = new SimpleProfile(list);
+        SimpleProfile profile = new SimpleProfile(Collections.<Attribute<?>>emptyList());
 
-        profile.getAttribute(null, String.class);
+        profile.getAttribute(null, Object.class);
     }
 
     @Test
-    public void getAttribute_shouldReturnStringValueForExistingKey() {
-        Profile profile = new SimpleProfile(asList(SOME_KEY, STRING_VALUE));
-
-        String result = profile.getAttribute(SOME_KEY);
+    public void getAttributeTyped_shouldReturnNullValueForNonExistingKey() {
+        SimpleProfile profile = new SimpleProfile(Collections.<Attribute<?>>emptyList());
 
-        assertEquals(STRING_VALUE, result);
+        assertNull(profile.getAttribute(SOME_KEY, Object.class));
     }
 
     @Test
-    public void getAttribute_shouldReturnNullValueForNonExistingKey() {
-        List<Attribute> list = Collections.emptyList();
-        Profile profile = new SimpleProfile(list);
+    public void getAttributeTyped_shouldReturnTypedValueForExistingKey() {
+        SimpleProfile profile = new SimpleProfile(asAttributeList(SOME_KEY, STRING_VALUE));
 
-        assertNull(profile.getAttribute(SOME_KEY));
-        assertNull(profile.getAttribute(SOME_KEY, Integer.class));
-    }
+        Attribute<String> result = profile.getAttribute(SOME_KEY, String.class);
 
-    @Test
-    public void getAttribute_shouldReturnNullValueForMismatchingType() {
-        Profile profile = new SimpleProfile(asList(SOME_KEY, INTEGER_VALUE));
-
-        assertNull(profile.getAttribute(SOME_KEY));
-        assertNull(profile.getAttribute(SOME_KEY, String.class));
+        assertEquals(STRING_VALUE, result.getValue());
     }
 
     @Test
-    public void getAttribute_shouldReturnIntegerValueForExistingKey() {
-        Profile profile = new SimpleProfile(asList(SOME_KEY, INTEGER_VALUE));
+    public void getAttributeTyped_shouldThrowExceptionForForMismatchingType() {
+        SimpleProfile profile = new SimpleProfile(asAttributeList(SOME_KEY, INTEGER_VALUE));
 
-        Integer result = profile.getAttribute(SOME_KEY, Integer.class);
-
-        assertEquals(INTEGER_VALUE, result);
+        try {
+            profile.getAttribute(SOME_KEY, String.class);
+        } catch (ClassCastException e) {
+            assertTrue(e.getMessage().contains("java.lang.String"));
```

Comment:
Is it good to rely on the message for the test assertion? It's not better to catch the exception and check if it is of the type that we wanted?

Reply:
It does catch ClassCastException, which is what we are looking for.  
Checking the message was just to ensure I'd used the correct the method on Class (.getCanonicalName())

Reply:
then you can use `String.class.getName()`